### ISSUE TITLE
Help to support tensorflow slim "Flatten" pattern to Tensorrt.

### ISSUE THREAD

**Issue by ys0232** [Tencent/Forward#13]
There is a layer in tensorflow slim named "Flatten", it includes servel tensorflow operations like: "Shape", "StridedSlice" and "Reshape". The test code like this.

```
import numpy as np
import os


def create_tf_flatten(model_file):
    import tensorflow as tf
    import tf_slim as slim
    with tf.Session() as sess:
        x1 = tf.placeholder(shape=(None,299,299,3),dtype=tf.float32, name='x')

        op = slim.flatten(x1)

        sess.run(tf.global_variables_initializer())
        feed_dict = {x1: np.ones((1,299,299,3))}
        print(sess.run(op, feed_dict))

        graphdef = tf.graph_util.convert_variables_to_constants(sess, sess.graph_def, ['Flatten/flatten/Reshape'])
        tf.train.write_graph(graphdef, './', model_file, as_text=False)
        return feed_dict

def forward_transfer(model_file, dummy_input):
    import forward
    # 1. 构建 Engine
    builder = forward.TfBuilder()
    infer_mode = 'float32'

    builder.set_mode(infer_mode)
    tf_engine = builder.build(model_file, dummy_input)

    # save engine
    engine_path = os.path.splitext(model_file)[0] + '.engine'
    tf_engine.save(engine_path)


def test_forward(model_file, dummy_input):
    import forward
    engine_path = os.path.splitext(model_file)[0] + '.engine'
    # load saved engine
    tf_engine = forward.TfEngine()
    tf_engine.load(engine_path)

    inputs = dummy_input
    outputs = tf_engine.forward(inputs)
    print(outputs)


model_file = 'tf_model.pb'
create_tf_flatten(model_file)

x = {'x':np.ones([1,299,299,3],dtype='float32')}

forward_transfer(model_file, x)
test_forward(model_file, x)
```
## Environment

**TensorRT Version**: 7.1.3.4
**NVIDIA GPU**: T4
**NVIDIA Driver Version**: 410.104
**CUDA Version**: 10.2
**CUDNN Version**: 8.0
**Operating System**: ubuntu 18.04
**Python Version (if applicable)**: 3.6.9
**Tensorflow Version (if applicable)**: 1.15.0
**PyTorch Version (if applicable)**: 1.7.0

@aster2013 @yuanzexi

**Comment by ys0232** [comment]
Another unsupport operation is tf.depth_to_space, it in torch is "PixelShuffle". Thanks for helping!

**Comment by yuanzexi** [comment]
@ys0232  In the latest commit eb0eb949caa13009c166bbebdddad064a9bf4767, Forward can support 'Flatten' Op. You could try it. And we'll support `tf.depth_to_space` soon.

**Comment by ys0232** [comment]
Nice work! @yuanzexi

**Comment by ys0232** [comment]
There are another "Reshape" pattern like "Flatten". However, it doesn't work in Forward-1.1.0. Please help to support "Reshape" to tensorrt, thanks! Create model code look like this.@yuanzexi
```
def create_tf_reshape(model_file):
    import tensorflow as tf
    import tf_slim as slim
    with tf.Session() as sess:
        x1 = tf.placeholder(shape=(1,None,None,27),dtype=tf.float32, name='x')

        #op = slim.flatten(x1)
        gs = tf.shape(x1)
        op = tf.reshape(x1,shape=(1,gs[1],gs[2],3,9),name='reshape')

        sess.run(tf.global_variables_initializer())
        feed_dict = {x1: np.ones((1,12,12,27))}
        print(sess.run(op, feed_dict))

        graphdef = tf.graph_util.convert_variables_to_constants(sess, sess.graph_def, ['reshape'])
        tf.train.write_graph(graphdef, './', model_file, as_text=False)
        return feed_dict
```

**Comment by yuanzexi** [comment]
> There are another "Reshape" pattern like "Flatten". However, it doesn't work in Forward-1.1.0. Please help to support "Reshape" to tensorrt, thanks! Create model code look like this.@yuanzexi
> 
> ```
> def create_tf_reshape(model_file):
>     import tensorflow as tf
>     import tf_slim as slim
>     with tf.Session() as sess:
>         x1 = tf.placeholder(shape=(1,None,None,27),dtype=tf.float32, name='x')
> 
>         #op = slim.flatten(x1)
>         gs = tf.shape(x1)
>         op = tf.reshape(x1,shape=(1,gs[1],gs[2],3,9),name='reshape')
> 
>         sess.run(tf.global_variables_initializer())
>         feed_dict = {x1: np.ones((1,12,12,27))}
>         print(sess.run(op, feed_dict))
> 
>         graphdef = tf.graph_util.convert_variables_to_constants(sess, sess.graph_def, ['reshape'])
>         tf.train.write_graph(graphdef, './', model_file, as_text=False)
>         return feed_dict
> ```

So far, Forward only support DYNAMIC BATCH dim, so using `shape=(1, None, None, 27)` as input is invalid.

**Comment by ys0232** [comment]
> > There are another "Reshape" pattern like "Flatten". However, it doesn't work in Forward-1.1.0. Please help to support "Reshape" to tensorrt, thanks! Create model code look like this.@yuanzexi
> > ```
> > def create_tf_reshape(model_file):
> >     import tensorflow as tf
> >     import tf_slim as slim
> >     with tf.Session() as sess:
> >         x1 = tf.placeholder(shape=(1,None,None,27),dtype=tf.float32, name='x')
> > 
> >         #op = slim.flatten(x1)
> >         gs = tf.shape(x1)
> >         op = tf.reshape(x1,shape=(1,gs[1],gs[2],3,9),name='reshape')
> > 
> >         sess.run(tf.global_variables_initializer())
> >         feed_dict = {x1: np.ones((1,12,12,27))}
> >         print(sess.run(op, feed_dict))
> > 
> >         graphdef = tf.graph_util.convert_variables_to_constants(sess, sess.graph_def, ['reshape'])
> >         tf.train.write_graph(graphdef, './', model_file, as_text=False)
> >         return feed_dict
> > ```
> 
> @ys0232 So far, Forward only support DYNAMIC BATCH dim, so using `shape=(1, None, None, 27)` as input is invalid.

**Comment by ys0232** [comment]
> Another unsupport operation is tf.depth_to_space, it in torch is "PixelShuffle". Thanks for helping!

@yuanzexi Could help to support tf.depth_to_space and pytorch.PixelShuffle ?

**Comment by yuanzexi** [comment]
@ys0232  Sorry for  late reply. tf.depth_to_space and pytorch.PixelShuffle are now supported in [Release v1.2.2](https://github.com/Tencent/Forward/releases/tag/v1.2.2).
